nodejs/week2/app.js

```diff
@@ -0,0 +1,80 @@
+// @ts-ignore
+const express = require('express')
+const router = express.Router
+const app = express()
+const port = process.env.PORT || 3000
+const document = require("./documents.json")
+
+
+// Support parsing JSON requests
+app.use(express.json())
+
+// 1:GET /search  ex: GET /search?q=hello:
+app.get('/search',  (req, res) => {
+    try{
+    // @ts-ignore
+    const result =  document.filter(element => Object.values(element).find(word => String(word).includes(req.query.q)));
+   if(result){
+    res.send(result)
+}else{
+    res.sendStatus(404).json({message: "Record not found"});
+}
+}catch(error){
+    res.send(error)
+}
+})
+  
+// 2:GET /documents/:id
+// @ts-ignore
+app.get('/documents/:id', (req, res) => {
+    try{
+    var result =  document.filter(it => it.id === Number(req.params.id));
+    if(result){
+    res.send(result);
+}else{
+    res.sendStatus(404).json({message: "Record not found"});
```

Comment:
Use
`    res.status(404).json({message: "Record not found"});
`
Instead.
res.sendStatus only sends status but not json. And sendStatus doesnt break the flow meaning catch/throw will again try to post status back to client that may fail.